File: frontend/src/app/services/category.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class CategoryService {

  constructor(private _http:HttpClient) { }

  // Load all the categories
  public categories(){
    return this._http.get(`${baseUrl}/category/`);
  }

  // add new category
  public addCategory(category:any){
    return this._http.post(`${baseUrl}/category/`, category);
  }

  // delete category
  public deleteCategory(cid:any){
    return this._http.delete(`${baseUrl}/category/${cid}`);
  }

  // get one category
  public getCategory(cid:any){
    return this._http.get(`${baseUrl}/category/${cid}`);
  }

  // update category
  public updateCategory(category:any){
    return this._http.put(`${baseUrl}/category/`, category);
  }
  
}

File: backend/src/main/java/com/classstudies/classquiz/service/CategoryService.java
package com.classstudies.classquiz.service;

import java.util.Set;

import com.classstudies.classquiz.model.exam.Category;

public interface CategoryService {
	
	public Category addCategory(Category category);
	
	public Category updateCategory(Category category);
	
	public Set<Category> getCategories();
	
	public Category getCategory(Long categoryId);
	
	public void deleteCategory(Long categoryId);

}
